src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -1127,14 +1129,18 @@ internal EntityHandle TryGetTypeHandle(Type type)
                 return eb._typeBuilder._handle;
             }
 
-            if (IsConstructedFromTypeBuilder(type))
+            if (IsConstructedFromTypeBuilder(type) ||
+                HasElementTypeOfTypeBuilderOrConstructedFromTypeBuilder(type))
             {
                 return default;
             }
 
             return GetTypeReferenceOrSpecificationHandle(type);
         }
 
+        private static bool HasElementTypeOfTypeBuilderOrConstructedFromTypeBuilder(Type type) => type.HasElementType &&
+            (type.GetElementType() is TypeBuilderImpl || IsConstructedFromTypeBuilder(type.GetElementType()!));
```

Comment:
I think it would be best to make this condition part of IsConstructedFromTypeBuilder method instead.

Reply:
> I do not think this handles jagged arrays correctly, like: MyRefEmittedType[][][].

How can I create a jagged array with Ref.Emit? `typeBuilder.MakeArrayType(3)` creates `MyType[,,]` not `MyType[][][]`. FWIW it works for `MyType[,,]`.

> I think it would be best to make this condition part of IsConstructedFromTypeBuilder method instead.

Indeed, moved, thanks

Reply:
> How can I create a jagged array with Ref.Emit? 

`typeBuilder.MakeArrayType().MakeArrayType().MakeArrayType()`

Reply:
Thank you! Added test and it was indeed failing previously